lib/api/apiUtils/object/objectLockHelpers.js

```diff
@@ -158,6 +158,33 @@ function isObjectLocked(bucket, objectMD, headers) {
     return false;
 }
 
+
+/* objectLockRequiresBypass will return true if the retention info change
+ * would require a bypass governance flag to be true.
+ * In order for this to be true the action must be valid as well, so going from
+ * COMPLIANCE to GOVERNANCE would return false unless it expired.
+ */
+function objectLockRequiresBypass(objectMD, retentionInfo) {
+    const { retentionMode: existingMode, retentionDate: existingDateISO } = objectMD;
+    if (!existingMode) {
+        return false;
+    }
+
+    const existingDate = new Date(existingDateISO);
+    const isExpired = existingDate < Date.now();
+    const isExtended = new Date(retentionInfo.date) > existingDate;
+
+    if (existingMode === 'GOVERNANCE' && !isExpired) {
+        if (retentionInfo.mode === 'GOVERNANCE' && isExtended) {
+            return false;
+        }
+        return true;
+    }
+
+    // an invalid object lock state
```

Comment:
Can this comment expand on what are the invalid object lock states?

Reply:
I think more appropriately the comment should say:
```suggestion
    // an invalid retention change or unrelated to bypass
```
There are a few state changes that are invalid which would be caught by the validate later, or are irrelevant to the bypass we don't care about. One example above is going from GOVERNANCE to COMPLIANCE which does not require a bypass. Moving from COMPLIANCE to GOVERNANCE is not allowed, I can't quite remember what other before/after states are invalid.